dot/state/epoch.go

```diff
@@ -191,21 +191,25 @@ func (s *EpochState) GetEpochForBlock(header *types.Header) (uint64, error) {
 		return 0, errors.New("header is nil")
 	}
 
-	firstSlot, err := s.baseState.loadFirstSlot()
+	if header.Number == 1 {
+		return 0, nil
+	}
+
+	chainFirstSlotNumber, err := s.retrieveFirstNonOriginBlockSlot(header.Hash())
 	if err != nil {
-		return 0, err
+		return 0, fmt.Errorf("retrieving very first slot number: %w", err)
 	}
 
 	slotNumber, err := header.SlotNumber()
 	if err != nil {
 		return 0, fmt.Errorf("getting slot number: %w", err)
 	}
 
-	return (slotNumber - firstSlot) / s.epochLength, nil
+	return (slotNumber - chainFirstSlotNumber) / s.epochLength, nil
```

Comment:
@kishansagathiya `chainFirstSlotNumber` is the very first slot number that appear in the chain, it is also known as the slot number of block number `#1`, this slot number is important to calculate epochs correctly, I don't want to use `firstSlot` because we use this nomeclature to define the first slot of an epoch

Reply:
ok